src/shared/tpm2-util.c

```diff
@@ -2333,3 +2350,20 @@ int pcr_mask_to_string(uint32_t mask, char **ret) {
         *ret = TAKE_PTR(buf);
         return 0;
 }
+
+void tpm2_util_pbkdf(const void *key,
+                    size_t key_size,
+                    const void *input,
+                    size_t input_size,
+                    int iterations,
+                    uint8_t res[static SHA256_DIGEST_SIZE]) {
+
+        const void *salt = input;
+        size_t salt_size = input_size;
+        do {
+                hmac_sha256(key, key_size, salt, salt_size, res);
+                salt = res;
+                salt_size = SHA256_DIGEST_SIZE;
+        /* this requires signed logic */
+        } while (iterations-- > 0);
+}
```

Comment:
this isn't really related to tpm2 stuff so much, maybe move this to hmac.[ch], or maybe even create a new pbkdf.[ch] for this.

is this pbkdf2? if so, please add comment about that, maybe link to rfc…

Reply:
would be good to have quick test in src/test/test-pbkfd with some simple test vectors btw

Reply:
This is not PBKDF2, but I just implemented PBKDF2_HMACSHA256 and will update the PR with that.